src/server/scripts/EasternKingdoms/ZulGurub/boss_marli.cpp

```diff
@@ -77,142 +80,180 @@ enum Misc
+                                me->SetBaseWeaponDamage(BASE_ATTACK, MAXDAMAGE, (cinfo->maxdmg +((cinfo->maxdmg/100) * 1)));
+                                me->UpdateDamagePhysical(BASE_ATTACK);
+                                */
+                                me->ApplyStatPctModifier(UNIT_MOD_DAMAGE_MAINHAND, TOTAL_PCT, DamageDecrease); // hack
+                                events.ScheduleEvent(EVENT_ASPECT_OF_MARLI, 12s, 0, PHASE_TWO);
+                                events.ScheduleEvent(EVENT_TRANSFORM, 45s, 0, PHASE_TWO);
+                                events.ScheduleEvent(EVENT_POISON_VOLLEY, 15s);
+                                events.ScheduleEvent(EVENT_HATCH_SPIDER_EGG, 12s, 17s);
+                                events.ScheduleEvent(EVENT_TRANSFORM, urand(35000, 60000), 0, PHASE_TWO);
+                                events.SetPhase(PHASE_TWO);
+                                break;
                             }
-                            events.ScheduleEvent(EVENT_CHARGE_PLAYER, 8s, 0, PHASE_THREE);
-                            break;
-                        }
-                        case EVENT_TRANSFORM_BACK:
-                        {
-                            me->RemoveAura(SPELL_SPIDER_FORM);
-                            /*
-                            CreatureTemplate const* cinfo = me->GetCreatureTemplate();
-                            me->SetBaseWeaponDamage(BASE_ATTACK, MINDAMAGE, (cinfo->mindmg +((cinfo->mindmg/100) * 1)));
-                            me->SetBaseWeaponDamage(BASE_ATTACK, MAXDAMAGE, (cinfo->maxdmg +((cinfo->maxdmg/100) * 1)));
-                            me->UpdateDamagePhysical(BASE_ATTACK);
-                            */
-                            me->ApplyStatPctModifier(UNIT_MOD_DAMAGE_MAINHAND, TOTAL_PCT, DamageDecrease); // hack
-                            events.ScheduleEvent(EVENT_ASPECT_OF_MARLI, 12s, 0, PHASE_TWO);
-                            events.ScheduleEvent(EVENT_TRANSFORM, 45s, 0, PHASE_TWO);
-                            events.ScheduleEvent(EVENT_POISON_VOLLEY, 15s);
-                            events.ScheduleEvent(EVENT_SPAWN_SPIDER, 30s);
-                            events.ScheduleEvent(EVENT_TRANSFORM, urand(35000, 60000), 0, PHASE_TWO);
-                            events.SetPhase(PHASE_TWO);
-                            break;
+                            default:
+                                break;
                         }
-                        default:
-                            break;
+
+                        if (me->HasUnitState(UNIT_STATE_CASTING))
+                            return;
                     }
 
-                    if (me->HasUnitState(UNIT_STATE_CASTING))
-                        return;
+                    DoMeleeAttackIfReady();
                 }
 
-                DoMeleeAttackIfReady();
-            }
+            private:
+                std::set<GameObject*> _hatchedEggs;
```

Comment:
never store pointers to other world objects in scripts - if the object despawns after you store it in here but before you use it again, *bad things will happen* (bad things means accessing freed memory and most likely a crash)

Reply:
Yes, thats exactly what it is happening, but then how am I suppose to store despawned objects in order to spawn them again then?

Reply:
in general you should save their ObjectGuids and get their concrete objects via ObjectAccessor

Reply:
Pretty sure it will be safe to just respawn ALL of them at reset

Reply:
Use something like this https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_buru.cpp#L77-L86